src/android/de/martinreinhardt/cordova/plugins/hotspot/HotSpotPlugin.java

```diff
@@ -353,6 +355,20 @@ public void run(JSONArray args, CallbackContext callback) throws Exception {
       return true;
     }
 
+    if ("isConnectedToInternetViaEthernet".equals(action)) {
+      threadhelper(new HotspotFunction() {
+        @Override
+        public void run(JSONArray args, CallbackContext callback) throws Exception {
+          if (isConnectedToInternetViaEthernet()) {
+            callback.success();
+          } else {
+            callback.error("Device is not connected to internet via Ethernet");
```

Comment:
I think that throwing an error here is not a good idea since is not an error, why not just to return success with a false value?
Something like `callback.success(true)` or `callback.success(false)`?

Reply:
Good point. Will change that.